dataloader.py

```diff
@@ -0,0 +1,103 @@
+# PyTorch imports
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.sampler import SubsetRandomSampler
+
+# Other libraries for data manipulation and visualization
+import os
+import numpy as np
+
+
+class PadSequence:
+    def __call__(self, batch):
+        #Each element in batch is (features, labels)
+        sorted_batch = sorted(batch, key=lambda x: x[0].shape[0], reverse=True)
+
+        sequences = [x[0] for x in sorted_batch]
+        sequences_padded = torch.nn.utils.rnn.pad_sequence(sequences, batch_first=True)
+
+        labels = [x[1] for x in sorted_batch]
+        labels_padded = torch.nn.utils.rnn.pad_sequence(labels, batch_first=True)
+        return sequences_padded, labels_padded
+
+class PreprocessedParsedReplayDataset(Dataset):
+    """
+    Customed preprocessed input files for parsed game replays.
+    Features/labels should be in folder ./data/feautures and ./data/labels respectively, unless specified.
+    Each txt file in the folder is for 1 game.
+    """
+    
+    def __init__(self, feature_folder='./data/features/', label_folder='./data/labels/'):
+        
+        self.feature_folder = feature_folder
+        self.label_folder = label_folder
+        assert len(os.listdir(self.feature_folder))==len(os.listdir(self.label_folder)), "Number of files in feature and label folders do not match."
+        self.num_games = len(os.listdir(self.feature_folder))
+        
+    def __len__(self):
+        return self.num_games
+    
+    def __getitem__(self, ind):
+        
+        feature_path = os.path.join(self.feature_folder,str(int(ind))+'.txt')
+        label_path = os.path.join(self.label_folder,str(int(ind))+'.txt')
+        
+        features = np.loadtxt(feature_path)
+        labels = np.loadtxt(label_path)
+        
+        return torch.Tensor(features), torch.Tensor(labels)
```

Comment:
This is a very minor and subtle point: This is returning features and labels in a specific order. If someone were to use the dataloader, and then gets back an object, they would HAVE to know the order of features and labels.

I.e. You could use a dictionary instead:
```
return {"features": torch.Tensor(features), "labels": torch.Tensor(labels)}
```
This way, all someone needs to know is that there will be features and labels, not the order. This is particularly challenging to read since starting at the notebook file where it's used (where I start reading), I have to go 3 functions down to see what's being returned. You can think of the dictionary acting like a mini-class to organize the results with keywords.